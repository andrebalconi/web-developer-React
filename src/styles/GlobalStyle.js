import React from "react";

import { createGlobalStyle } from "styled-components";
import { BreakpointSize, breakAt } from "./Breakpoints";
import { Helmet } from "react-helmet";

const colorYellow = "#ffc107";

const GlobalStyle = createGlobalStyle`
    
    //@import url("https://fonts.googleapis.com/css2?family=Poppins:wght@300;600;700&display=swap");
    html {
        font-family: "Poppins", sans-serif;
        font-weight: 300;
        font-size: 100%;
        color: #212121;
        box-sizing: border-box;
    }

    *, *::before, *::after {
        box-sizing: inherit;
    }

    body {
        margin: 0;
    }

    h1, h2, h3, h4, h5, h6 {
        line-height: 1.3;
        font-weight: 700;
        letter-spacing: 2px;

        strong {
            color: ${colorYellow};
        }
    }

    h1 {
        font-size: 2.5rem;
        ${breakAt(BreakpointSize.lg)} {
            font-size: 3.75rem;
        }
    }

    h2 {
        font-size: 2rem;
        ${breakAt(BreakpointSize.lg)} {
            font-size: 3.125rem;
        }
    }

    h3 {
        font-size: 1.9rem;
        ${breakAt(BreakpointSize.lg)} {
            font-size: 2.5rem;
        }
    }
    
    h4 {
        font-size: 1.3rem;
        font-weight: 600;
        ${breakAt(BreakpointSize.lg)} {
            font-size: 2.125rem;
        }
    }

    h5 {
        font-size: 1.2rem;
        font-weight: 300;
        ${breakAt(BreakpointSize.lg)} {
            font-size: 1.5rem;
        }
    }

    h6 {
        font-size: 1.1rem;
        font-weight: 600;
        ${breakAt(BreakpointSize.lg)} {
            font-size: 1.25rem;
        }
    }

`;

const GlobalStyleComposed = () => (
  <>
    <GlobalStyle />
    <Helmet>
    <link
      href="https://fonts.googleapis.com/css2?family=Poppins:wght@300;600;700&display=swap"
      rel="stylesheet"
    />
    </Helmet>
  </>
);

export default GlobalStyleComposed;
