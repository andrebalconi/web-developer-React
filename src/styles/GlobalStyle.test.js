import React from "react";
import { render } from "@testing-library/react";

import GlobalStyle from "./GlobalStyle";

test("match snapshot", () => {
  render(<GlobalStyle />);

  expect(document.head).toMatchSnapshot();
});
