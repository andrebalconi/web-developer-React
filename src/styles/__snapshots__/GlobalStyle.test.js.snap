// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<head>
  <style
    data-styled="active"
    data-styled-version="5.3.6"
  />
</head>
`;
